api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -124,5 +130,9 @@ emitLogRecord Logger {..} LogRecordArguments {..} = do
       , logRecordBody = body
       , logRecordResource = loggerProviderResource loggerProvider
       , logRecordInstrumentationScope = loggerInstrumentationScope
-      , logRecordAttributes = attributes
+      , logRecordAttributes =
+          addAttributes
+            (loggerProviderAttributeLimits loggerProvider)
+            emptyAttributes
+            attributes
```

Comment:
Is there anything about this that addresses the part of the spec that says this:
> There SHOULD be a message printed in the SDK’s log to indicate to the user that an attribute was discarded due to such a limit. To prevent excessive logging, the message MUST be printed at most once per LogRecord (i.e., not per discarded attribute).

Reply:
This is actually something that we don't support on the tracing side (which says effectively the same thing). I wanted to support it on the tracing side using OTel logging, so it's a bit of an ouroboros 😆 

I think enough of the logging interface is in place that we should be be able to support this use case now though.

Reply:
This is addressed by the next upstream pull request in the stack. The change was to large to be reasonably contained in this PR.